src/backend/d3d12/QueueD3D12.cpp

```diff
@@ -32,9 +32,21 @@ namespace d3d12 {
             IID_PPV_ARGS(&commandList)
         ));
         ASSERT_SUCCESS(commandList->Close());
+
+        ASSERT_SUCCESS(device->GetD3D12Device()->CreateFence(fenceValue, D3D12_FENCE_FLAG_NONE, IID_PPV_ARGS(&fence)));
+        fenceEvent = CreateEvent(nullptr, FALSE, FALSE, nullptr);
+        ASSERT(fenceEvent != nullptr);
     }
 
     void Queue::Submit(uint32_t numCommands, CommandBuffer* const * commands) {
+        const uint64_t currentFence = fenceValue++;
```

Comment:
nit: can you add a TODO saying that this stalls on the previous submit but we should fix it at some point?